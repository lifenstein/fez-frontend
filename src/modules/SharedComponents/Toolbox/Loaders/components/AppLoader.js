import React from 'react';
import PropTypes from 'prop-types';
import { styled } from '@mui/material/styles';

import CircularProgress from '@mui/material/CircularProgress';
import Typography from '@mui/material/Typography';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';

const StyledGrid = styled(Grid)(({ theme }) => ({
    flexDirection: 'column',
    justifyContent: 'center',
    display: 'flex',
    alignItems: 'center',
    ...theme.palette.primary.gradient.diagonal,
    width: '100%',
    height: '100%',
    textAlign: 'center !important',
}));

export class AppLoader extends React.Component {
    static propTypes = {
        title: PropTypes.string.isRequired,
        logoImage: PropTypes.string,
        logoText: PropTypes.string,
    };

    render() {
        const { title, logoImage, logoText } = this.props;
        return (
            <StyledGrid container spacing={0} sx={{}}>
                <Grid item m={'16px 0'}>
                    <CircularProgress
                        size={80}
                        thickness={1}
                        sx={{ color: 'white.main', fontWeight: 'fontWeightLight' }}
                    />
                </Grid>
                <Grid item m={'16px 0'}>
                    {logoImage && <Box className={`${logoImage}`} sx={{ width: '200px' }} alt={logoText} />}
                </Grid>
                <Grid item m={'16px 0'}>
                    <Typography variant={'h6'} sx={{ color: 'white.main', fontWeight: 'fontWeightLight' }}>
                        {title}
                    </Typography>
                </Grid>
            </StyledGrid>
        );
    }
}

export default AppLoader;
