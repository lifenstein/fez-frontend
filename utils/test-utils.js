/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Route } from 'react-router';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';
import { createMemoryHistory } from 'history';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');
const mime = require('mime-types');

const { configure } = domTestingLib;

configure(config => ({
    ...config,
    testIdAttribute: 'id',
}));

export const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <MuiPickersUtilsProvider utils={MomentUtils}>{props.children}</MuiPickersUtilsProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }), renderMethod = render } = {},
) => {
    return renderMethod(
        <AllTheProviders>
            <Router history={history}>{ui}</Router>
        </AllTheProviders>,
    );
};

export const WithRouter = ({ children, route = '/', history = createMemoryHistory({ initialEntries: [route] }) }) => (
    <AllTheProviders>
        <Router history={history}>{children}</Router>
    </AllTheProviders>
);
export const renderWithRedux = ({ initialState }) => render => {
    return {
        ...render,
        store: getStore({ initialState, history: render.history }),
    };
};

export const withRouter = ({
    route = '/',
    path = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
} = {}) => WrappedComponent => {
    return (
        <Router history={history}>
            <Route path={path} children={WrappedComponent} />
        </Router>
    );
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

export const defaultDatastream = {
    dsi_embargo_date: null,
    dsi_open_access: 1,
    dsi_label: '',
    dsi_copyright: null,
    dsi_state: 'A',
    dsi_size: 27932352,
};
export const createFezDatastreamInfoArray = (datastreams, pid = null, withPreview = true) => {
    let processed = datastreams.filter(datastream => typeof datastream === 'object' || typeof datastream === 'string');
    if (withPreview) {
        processed = [];
        datastreams.forEach(function(datastream) {
            processed.push(datastream);
            const filename = typeof datastream === 'object' ? datastream.dsi_dsid : datastream;
            const mimetype = mime.lookup(filename);
            // bail in case it's not derivable
            if (
                !filename.includes('.pdf') &&
                !mimetype.includes('image') &&
                !mimetype.includes('audio') &&
                !mimetype.includes('video')
            ) {
                return;
            }
            // eslint-disable-next-line no-nested-ternary
            const derivativeMimetype = mimetype.includes('audio')
                ? 'audio/mp3'
                : mimetype.includes('video')
                ? 'video/mp4'
                : 'image/jpg';

            (derivativeMimetype.includes('image') ? ['preview_', 'thumbnail_', 'web_'] : ['']).forEach(function(
                prefix,
            ) {
                processed.push({
                    dsi_dsid: `${prefix}${filename.split('.')[0]}_t.${derivativeMimetype.split('/').pop()}`,
                    dsi_mimetype: derivativeMimetype,
                });
            });
        });
    }

    return processed.map((datastream, index) => ({
        ...defaultDatastream,
        ...{
            dsi_pid: pid,
            dsi_order: index + 1,
            dsi_mimetype: mime.lookup(typeof datastream === 'object' ? datastream.dsi_dsid : datastream),
        },
        ...(typeof datastream === 'object'
            ? datastream
            : {
                  dsi_dsid: datastream,
              }),
    }));
};

export const withDatastreams = (sources, datastreams, callback) =>
    datastreams.map(item => {
        const source = sources.find(datastream =>
            datastream.dsi_dsid.includes(
                item.dsi_dsid
                    .replace('_t.', '.')
                    .replace('preview_', '')
                    .replace('thumbnail_', '')
                    .replace('web_', '')
                    .split('.')[0],
            ),
        );
        const isDerivative = item.dsi_dsid !== source.dsi_dsid;
        callback(item, source, isDerivative);
    });

export const getDatastreamByFilename = (filename, derivatives) =>
    derivatives.find(derivative => derivative.dsi_dsid === filename);

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    renderWithRouter,
    renderWithRedux,
    withRedux,
    withRouter,
    AllTheProviders,
    WithReduxStore,
    WithRouter,
    createFezDatastreamInfoArray,
    getDatastreamByFilename,
    withDatastreams,
};
