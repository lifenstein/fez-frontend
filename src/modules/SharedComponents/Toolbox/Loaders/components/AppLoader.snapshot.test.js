import React from 'react';
import { AppLoader } from './AppLoader';
import { rtlRender } from 'test-utils';

function setup(testProps = {}) {
    const props = {
        ...testProps,
    };
    return rtlRender(<AppLoader {...props} />);
}

describe('Component AppLoader', () => {
    it('should render as expected', () => {
        // set initial props values
        const props = {
            title: 'Fez frontend',
            logoImage: 'http://image/image.svg',
            logoText: 'Fez logo',
        };
        const { container } = setup({ ...props });
        expect(container).toMatchSnapshot();
    });
});
