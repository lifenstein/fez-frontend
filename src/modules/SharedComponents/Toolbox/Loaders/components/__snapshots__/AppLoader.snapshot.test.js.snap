// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component AppLoader should render as expected 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column display alignItems justifyContent backgroundColor width height textAlign css-1ewlecp-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item margin css-13i4rnv-MuiGrid-root"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary color fontWeight css-u4ps4u-MuiCircularProgress-root"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="21.5"
            stroke-width="1"
          />
        </svg>
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item margin css-13i4rnv-MuiGrid-root"
    >
      <div
        alt="Fez logo"
        class="http://image/image.svg undefined"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item margin css-13i4rnv-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 color fontWeight css-1uywuy6-MuiTypography-root"
      >
        Fez frontend
      </h6>
    </div>
  </div>
</div>
`;

exports[`Component AppLoader should render with styles 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column AppLoader-appLoader-1 css-1ewlecp-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item AppLoader-spaceBetween-3 css-13i4rnv-MuiGrid-root"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary AppLoader-white-2 css-u4ps4u-MuiCircularProgress-root"
        role="progressbar"
        style="width: 80px; height: 80px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="21.5"
            stroke-width="1"
          />
        </svg>
      </span>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item AppLoader-spaceBetween-3 css-13i4rnv-MuiGrid-root"
    />
    <div
      class="MuiGrid-root MuiGrid-item AppLoader-spaceBetween-3 css-13i4rnv-MuiGrid-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 AppLoader-white-2 css-1uywuy6-MuiTypography-root"
      >
        Test loader
      </h6>
    </div>
  </div>
</div>
`;
